pkg/verify/signature.go

```diff
@@ -136,8 +164,12 @@ func verifyEnvelope(verifier signature.Verifier, envelope EnvelopeContent) error
 }
 
 func verifyEnvelopeWithArtifact(verifier signature.Verifier, envelope EnvelopeContent, artifact io.Reader) error {
-	err := verifyEnvelope(verifier, envelope)
-	if err != nil {
+	artifacts := []io.Reader{artifact}
```

Comment:
We could just remove `verifyEnvelopeWithArtifact ` and `verifyEnvelopeWithArtifactDigest ` altogether and call `verifyEnvelopeWithArtifacts` and `verifyEnvelopeWithArtifactDigests` directly in the `VerifySignatureWithArtifact` and `VerifySignatureWithArtifactDigest` functions in signed_entity.go.